pkg/util/ranger/detacher.go

```diff
@@ -714,8 +731,12 @@ func ExtractEqAndInCondition(sctx *rangerctx.RangerContext, conditions []express
 		if ma == nil {
 			if accesses[i] != nil {
 				if allEqOrIn(accesses[i]) {
-					newConditions = append(newConditions, accesses[i])
 					columnValues[i] = extractValueInfo(accesses[i])
+					if columnValues[i] != nil && columnValues[i].value != nil && columnValues[i].value.IsNull() {
+						accesses[i] = nil
+					} else {
+						newConditions = append(newConditions, accesses[i])
+					}
```

Comment:
With these code blocks, we could add some test cases like `where a is null or a in (1, 2, 3)` for `TestConvertRangeToPoint` in `tests/integrationtest/t/planner/core/casetest/integration.test`

The `points2EqOrInCond` called in line#737 should also be updated?

Reply:
Hi @winoros thanks for the comments.

> With these code blocks, we could add some test cases like `where a is null or a in (1, 2, 3)` for `TestConvertRangeToPoint` in `tests/integrationtest/t/planner/core/casetest/integration.test`

Sure

> The `points2EqOrInCond` called in line#737 should also be updated?

I'm pretty sure it's a no-op to do this since intersecting a null point with non-null points works as expected, but I'll do it anyways to short-circuit the logic.